Updates for Astro Runtime Multiple Architecture Support (#1481)
astronomer/docs: astro/deploy-code.md

Follow the steps in this document to manually push your Astro project to a Deployment. For production environments, Astronomer recommends automating all code deploys with CI/CD. See [CI/CD](ci-cd.md).

> :::warning

Reviewer: I included this warning in my release notes as well. Do you think it's enough to just include it in prerequisites?

Author: @jwitz I duplicated the information so it couldn't be missed. Users scan text and there is a risk it could be missed. This is simply belt and suspenders.

Reviewer: @kwcanuck Given that this won't break anything in their system should they try to run without the right CLI version, I think it's okay to keep it in one place. I also downgraded the note from a warning to a caution because of this.

> :::warning

Reviewer: ```suggestion
:::caution
```

> If you're using an Apple M1 system for development, you must install Astro CLI 1.5.0 or later or your deploys will fail.

Reviewer: Only if your project uses Runtime 6.0.4+ right?

Author: @jwitz - good catch. I've updated the text.

> If you're using an Apple M1 system with Astro Runtime 6.0.4 or later for development, you must install Astro CLI 1.5.0 or later or your deploys will fail.

Reviewer 2: ```suggestion
If you're using Astro Runtime 6.0.5 or later on a Mac computer with the Apple M1 chip, you must install Astro CLI 1.4.0 or later or your deploys will fail. See [Install the CLI](cli/install-cli.md).
```

:::

## Prerequisites

> - The [Astro CLI](cli/overview.md) installed in an empty directory. If you're using an Apple M1 system with Astro Runtime 6.0.4 or later for development, you must install Astro CLI 1.5.0 or later.

Reviewer: ```suggestion
- The [Astro CLI](cli/overview.md) installed in an empty directory. If you're using an Apple M1 system with Astro Runtime 6.0.4 or later for local development, you must install Astro CLI 1.5.0 or later to deploy your work to Astro.
```

> - The [Astro CLI](cli/overview.md) is installed in an empty directory. If you're using an Apple M1 system with Astro Runtime 6.0.4 or later for local development, you must install Astro CLI 1.5.0 or later to deploy your work to Astro.

Reviewer 2: Do we need to have this in here twice? (I see the warning above)
